contracts/governance/StkTruToken.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import {VoteToken} from "./VoteToken.sol";
+import {ClaimableContract} from "../trusttoken/common/ClaimableContract.sol";
+import {ITruPriceOracle} from "./interface/ITruPriceOracle.sol";
+
+contract StkTruToken is VoteToken, ClaimableContract, ReentrancyGuard {
+    using SafeMath for uint256;
+
+    uint256 public constant MIN_STAKE_TIME = 14 days;
+
+    IERC20 public trustToken;
+    IERC20 public tusd;
+    ITruPriceOracle public oracle;
+    mapping(address => uint256) public unlockTime;
+
+    event OracleChanged(ITruPriceOracle newOracle);
+    event Stake(address indexed staker, uint256 amount, uint256 minted);
+    event Unstake(address indexed staker, uint256 burntAmount, uint256 truAmount, uint256 tusdAmount);
+
+    function initialize(
+        IERC20 _trustToken,
+        IERC20 _tusd,
+        ITruPriceOracle _oracle
+    ) public {
+        require(!initalized, "StkTruToken: Already initialized");
+        trustToken = _trustToken;
+        tusd = _tusd;
+        oracle = _oracle;
+        owner_ = msg.sender;
+        initalized = true;
+    }
+
+    function setOracle(ITruPriceOracle _oracle) external onlyOwner {
+        oracle = _oracle;
+        emit OracleChanged(_oracle);
+    }
+
+    function stake(uint256 amount) external {
+        unlockTime[msg.sender] = block.timestamp + MIN_STAKE_TIME;
+
+        uint256 amountToMint = totalSupply == 0 ? amount : totalSupply.mul(amount).div(value());
```

Comment:
I'm unsure if this is how we want to calculate amountToMint. Could you explain this line? Why are we using the value of the pool in USD to calculate how much to mint?

Reply:
We want to mint amount of tokens that would represent staker's share in the stake. That is
Before
Stake value (in TRU) = V
Supply = S
Amount (in TRU) = A
Minted = X
After:
Value = V+A
Supply = S+X

X/(S+X)*(V+A)=A
X(V+A)=AS+AX
XV+XA=SA+XA
XV=SA
X = SA/V